platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ForkJoinPoolHierarchicalTestExecutorServiceTests.java

```diff
@@ -48,17 +58,48 @@ void exceptionsFromInvalidConfigurationAreNotSwallowed() {
 		assertThat(exception).rootCause().isInstanceOf(IllegalArgumentException.class);
 	}
 
-	@Test
-	@Timeout(5)
-	void defersTasksWithIncompatibleLocks() throws Exception {
-		var configuration = new DefaultParallelExecutionConfiguration(2, 2, 2, 2, 1, __ -> true);
+	static List<Arguments> incompatibleLockCombinations() {
+		return List.of(//
+			arguments(//
+				Set.of(GLOBAL_READ), //
+				Set.of(GLOBAL_READ_WRITE) //
+			), //
+			arguments(//
+				Set.of(new ExclusiveResource("a", LockMode.READ)), //
+				Set.of(new ExclusiveResource("a", LockMode.READ_WRITE)) //
+			), //
```

Comment:
I'd also include examples of invalid multi-lock combinations.

e.g.
- `GR, aR, bR, dR` - `GR, aR, cR`

Reply:
Done in 5a88d210b7ad79bcd9935e5179fa97da50b16a31.